egs/wsj/s5/steps/nnet3/lstm/train.sh

```diff
@@ -451,6 +453,12 @@ while [ $x -lt $num_iters ]; do
   ilr=$initial_effective_lrate; flr=$final_effective_lrate; np=$num_archives_processed; nt=$num_archives_to_process;
   this_learning_rate=$(perl -e "print (($x + 1 >= $num_iters ? $flr : $ilr*exp($np*log($flr/$ilr)/$nt))*$this_num_jobs);");
 
+  if [ "$adaptive_shrink" == "true" ]; then
+    this_shrink=$(perl -e "print $shrink + ($flr*$this_num_jobs/$this_learning_rate) * (1 - $shrink)")
```

Comment:
The way you are doing it would make the initial shrinking rate independent of the num-jobs-initial, that's true, but it would make the final shrinking rate highly dependent on num-jobs-initial too.  So, for instance, between num-jobs-initial=1 and num-jobs-initial=2 there would be a factor of 2 difference in the final-probs.  I think it's more appropriate to do it the way I proposed, as in
 this_shrink=$(perl -e "print 1.0 - (($this_learning_rate / $ilr)) * (1 - $shrink));").
because this way, as we change the initial-num-jobs, the shrinking rate increases, which is actually appropriate because the initial actual learning rate is also increasing.  [this is in line with l2 regularization interpretation].